baseplate/features/__init__.py

```diff
@@ -0,0 +1,93 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+import logging
+
+from collections import namedtuple
+
+from .feature import feature_flag_from_config
+from ..file_watcher import FileWatcher, WatchedFileNotAvailableError
+from ..context import ContextFactory
+
+
+logger = logging.getLogger(__name__)
+
+
+class User(object):
+
+    def __init__(self, id, name, created, flags=None):
+        self.id = id
+        self.name = name
+        self.created = created
+        self.flags = flags or set()
+
+    @property
+    def logged_in(self):
+        return self.name is not None
+
+
+Content = namedtuple("Content", ["id", "type"])
+
+
+class FeatureFlagsContextFactory(ContextFactory):
+    """ ContextFactory for FeatureFlags.
+
+    This factory will attach a :py:class:`baseplate.features.FeatureFlags`
+    object on the :term:`context object` that will use the config file
+    specified by the given path.  Uses a :py:class:`baseplate.file_watcher.FileWatcher`
```

Comment:
probably best not to document how the internals work beyond what you want to promise forever and ever